### ISSUE TITLE
Changing IP-Address of NFS Server

### ISSUE THREAD

**Issue by inzanez** [democratic-csi/democratic-csi#377]
Hi

I just had to move an NFS server connected through democratic-csi to our Kubernetes cluster to a different network. Is there any easy way to change the IP address of the target server,...? Or what should be done in a case like this?

I assume it would be preferreable to use DNS names instead of IP addresses for storage targets,...that way it could be moved?

**Comment by travisghansen** [comment]
Yes DNS would be preferred. I have a tool to migrate them here: https://github.com/democratic-csi/democratic-csi/blob/master/contrib/pv-updater.sh

It is a fairly precarious process so please read through the code and review what it does. After that run it on a test pv and just make sure it's doing everything you need.

**Comment by inzanez** [comment]
Oh, I see. Yes, I will do so. But apart from changing the PVs (or to be more precise: delete them and re-create them),...I would also have to update the democratic-csi installation with a new server location. I guess this works without uninstall / re-install?
I just want to prevent existing PVCs from being deleted,...:-)

**Comment by travisghansen** [comment]
You can safely roll out the new config, the existing items won't be deleted unless you delete the respective pvc/pv. Assuming the nas is available at both IPs for a period of time the existing volumes will continue to mount etc and new volumes will get the new endpoint. When you're ready just start migrating the old PVs to the new endpoint and then eventually when ready you can remove the old IP from the nas entirely.

**Comment by inzanez** [comment]
For some reason I run into issues when making the NAS available at the second IP. Not quite sure what is happening, but all my PVs become inaccessible. Not sure if there is some asymmetric routing involved as the new NAS IP is in the same network as the Kubernetes workers. And so,...the responses might take a direct route,...? Although I was under the impression that this should not be an issue.
So maybe I'll have to shutdown everything and make the changes first,...
